### ISSUE TITLE
Prerendering error on routes whose filename start with same value as app.baseURL

### ISSUE THREAD

**Issue by yakir-reznik** [nuxt/nuxt#24389]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.1.0`
- Nuxt Version:     `3.8.2`
- CLI Version:      `3.10.0`
- Nitro Version:    `2.8.0`
- Package Manager:  `npm@9.6.6`
- Builder:          `-`
- User Config:      `app`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

When a value is set to **app.baseURL** option in **nuxt.config.ts** routes whose filename start with the app.baseURL throw an error when running **npm run generate** command.

[Link to reproduction repository](https://github.com/yakir-reznik/nuxt-app-base-bug)

### Describe the bug

# This repo has been made to reproduce a bug with app.baseURL in nuxt version 3.3.8

## Bug description
When a value is set to **app.baseURL** option in **nuxt.config.ts** routes whose filename start with the app.baseURL throw an error when running *npm run generate* command.

## Example
```bash
// nuxt.config.ts
export default defineNuxtConfig({
  app: {
    baseURL: "/some-folder/"
  }
})
```

Assume there is a file in **/pages/some-folder-page.vue

When you run *"npm run generate"* you will get this error (also attaching image): 
```bash
Errors prerendering:
  ├─ /some-folder-page (56ms)
  │ └── Error: [404] Cannot find any route matching -page.
```

![screenshot](https://github.com/yakir-reznik/nuxt-app-base-bug/blob/master/public/screenshot.png)

### Additional context

I have also tried defining the **app.baseURL** with and without slashes like "/value", "value/", "/value/" & "value".
The error happens in all cases.

.vite-inspect folder zip: [vite-inspect.zip](https://github.com/nuxt/nuxt/files/13427220/vite-inspect.zip)

**Comment by danielroe** [comment]
Would you see if you can reproduce with just nitro?

https://stackblitz.com/github/unjs/nitro/tree/main/examples/hello-world

**Comment by yakir-reznik** [comment]
@danielroe I tried, seems like it's a Nitro bug...
https://stackblitz.com/edit/github-lj8o8l?file=nitro.config.ts
Can you forward the issue to them or should i open another issue on the nitro repo?

**Comment by danielroe** [comment]
I mean, that would be very kind if you are able to do so 🙏

**Comment by yakir-reznik** [comment]
Actually, if i define nitro.config.ts like this (omitting the slashes):
```base
export default defineNitroConfig({
  baseURL: 'some-folder',
  prerender: {
    routes: ['/', 'some-folder-page'],
  },
});
```
Everything works as expected...
This leads me to believe the bug is related to how nuxt passes the config to Nitro...
@danielroe What do you think?

[Same stackblitz link →](https://stackblitz.com/edit/github-lj8o8l?file=nitro.config.ts)

**Comment by danielroe** [comment]
This is the correct config:

```ts
export default defineNitroConfig({
  baseURL: '/some-folder',
  prerender: {
    routes: ['/', '/some-folder-page'],
  },
});
```

And it is a Nitro (or radix3) bug.

**Comment by pi0** [comment]
@danielroe Can you please verify if radix3 1.1.1 solves this issue?

**Comment by danielroe** [comment]
No, it doesn't unfortunately. You can try in the playground with:

```ts
export default defineNuxtConfig({
  app: {
    baseURL: '/some-folder',
  },
  nitro: {
    prerender: {
      routes: ['/', '/some-folder-page'],
    }
  },
});
```

and `pnpm nuxi generate playground`.

The error thrown is:

```
ℹ Prerendering 4 initial routes with crawler                                                                          nitro 8:57:55 PM
  ├─ /some-folder-page (22ms)                                                                                          nitro 8:57:55 PM
  │ └── Error: [404] Cannot find any route matching -page.
  ├─ /200.html (9ms)                                                                                                   nitro 8:57:55 PM
  ├─ /404.html (4ms)                                                                                                   nitro 8:57:55 PM
  ├─ / (53ms)                                                                                                          nitro 8:57:55 PM
  ├─ /some-folder/_payload.json (1ms)                                                                                  nitro 8:57:55 PM
  ├─ /_payload.json (1ms)                                                                                              nitro 8:57:55 PM
                                                                                                                       nitro 8:57:55 PM
Errors prerendering:
  ├─ /some-folder-page (22ms)                                                                                          nitro 8:57:55 PM
  │ └── Error: [404] Cannot find any route matching -page.
```

**Comment by JokubasKuskysHostinger** [comment]
@danielroe Came over to the same issue, is there any suggested workaround for this or currently it is best that can be done is to not have baseUrl matching beginning of page route?

**Comment by danielroe** [comment]
on a fresh nuxt project, this is no longer reproducible - I think fixed upstream in `ufo`, likely by https://github.com/unjs/ufo/pull/313
